package/MDAnalysis/topology/DLPolyParser.py

```diff
@@ -129,16 +129,14 @@ def parse(self, **kwargs):
         with openany(self.filename) as inf:
             inf.readline()
             levcfg, imcon, megatm = np.int64(inf.readline().split()[:3])
-            inf.readline()
-            if not imcon == 0:
-                inf.readline()
-                inf.readline()
-                inf.readline()
 
             names = []
             ids = []
 
             line = inf.readline()
+            while not len(line.split()) == 5:
+                line = inf.readline()
```

Comment:
I think this has the ability to infinitely loop if a (non HISTORY) file never has 5 tokens on a line.  I think once we reach end-of-file `readline()` is going to return an empty string (i.e. not even a newline char), can we add a check for that and raise an error if we reach the end of file?

Reply:
It now raises EOFError, thanks!